blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,829 @@
+	cdnName?:     string | null;
+	domainName:   string;
+	guid?:        string | null;
+	hostName:     string;
+	httpsPort?:   number | null;
+	id?:          bigint;
+
+	// ILO things aren't being moved in because they also require a
+	// username/password that aren't used by any other interface type.
+	iloIpAddress?: string | null;
+	iloIpGateway?: string | null;
+	iloIpNetmask?: string | null;
+	iloPassword?:  string | null;
+	iloUsername?:  string | null;
+
+	/**
+	 * New field containing additional interfaces.
+	 * It is illegal for a server to not have at least one interface.
+	 */
+	interfaces: Array<Interface> & {0: Interface};
+	lastUpdated?:  string;
+
+	// notice not here: mgmtIpAddress, mgmtIpGateway, mgmtIpNetmask
+	// these will be moved into the interfaces property
+
+	offlineReason:   string;
+	physLocation?:   string | null;
+	physLocationId:  number;
+	profile?:        string | null;
+	profileDesc?:    string | null;
+	profileId:       number;
+	revalPending?:   boolean;
+	rack?:           string | null;
+	routerHostName?: string | null;
+	routerPortName?: string | null;
+	status?:         string | null;
+	statusId:        number;
+	tcpPort:         number;
+	type?:           string | null;
+	typeId:          number;
+	updPending?:     boolean;
+	xmppId?:         string | null;
+	xmppPassword?:   string | null;
+}
+```
+
+`/monitoring` payloads and CDN Snapshots are purposely un-versioned and
+un-modeled, so this section will make no attempts to define the model changes to
+those objects.
+
+## Impacted Components
+The only components that should need to be modified to accommodate this feature
+are Traffic Ops, Traffic Portal, and Traffic Monitor.
+
+### Traffic Ops Impact
+
+#### Database
+A new table will need to be introduced to contain interface information:
+`interface`, which is described below.
+```text
+            Table "traffic_ops.interface"
+     Column    |  Type   | Collation | Nullable | Default
+---------------+---------+-----------+----------+---------
+ name          | text    |           | not null |
+ server        | bigint  |           | not null |
+ monitor       | boolean |           | not null |
+ mtu           | integer |           |          | 1500
+ max_bandwidth | integer |           | not null | 0
+Indexes:
+    "interface_pkey" PRIMARY KEY, btree (name, server)
+Check constraints:
+	"interface_mtu_check" CHECK (mtu > 1280)
+Foreign-key constraints:
+    "interface_server_fkey" FOREIGN KEY (server) REFERENCES server(id)
+```
+Another new table will need to be created to contain address information, which
+is described below.
+```text
+               Table "traffic_ops.ip_address"
+     Column     |  Type   | Collation | Nullable | Default
+----------------+---------+-----------+----------+---------
+ address        | inet    |           | not null |
+ gateway        | inet    |           | not null |
+ serviceaddress | boolean |           | not null |
+ server         | bigint  |           | not null |
```

Comment:
How do we guarantee consistency between interface.server and ip_address.server/ip_address.interface?

Reply:
With a foreign key constraint. The primary key of an interface is `(name, server)`, so `(interface, server)` is a foreign key into interfaces.